src/test/ui/union-fields.stderr

```diff
@@ -0,0 +1,14 @@
+error: field is never used: `y`
+  --> $DIR/union-fields.rs:20:5
+   |
+20 |     y: u32,
+   |     ^^^^^^
```

Comment:
Diagnostics for unions are not actually tested here.
Could you remove the `struct V` (it's not especially relevant to the test) and test this instead:
```rust
union U1 {
    a: u8, // should not be reported
    b: u8, // should not be reported
    c: u8, // should be reported
}
union U2 {
    a: u8, // should be reported
    b: u8, // should not be reported
    c: u8, // should not be reported
}

let u = U1 { a: 0 };
let a = u.b;

let u = U2 { c: 0 };
let b = u.b;
```